Lib/traceback.py

```diff
@@ -132,7 +158,16 @@ def format_exception_only(etype, value):
     The message indicating which exception occurred is always the last
     string in the list.
 
+    Since python 3.7, instead of passing *etype* and *value* separately, it is
+    recommended to use the *exc* keyword argument which will infer all the
+    required values.
     """
+    if exc and (etype or value):
+        raise ValueError("Only one of `exc` or `(etype, value, tb)` can be set")
+    elif exc:
+        value = exc
+        tb = exc.__traceback__
+        etype = type(value)
```

Comment:
It seems like these lines are duplicated 3 times: write a private helper function to factorize the code.